### PR TITLE
Supermatter Delamination Balance Changes

### PR DESCRIPTION
**Pull request by Lufferly**
## About The Pull Request

The supermatter delamination countdown timer (how long it takes to go boom-boom after hitting 0 integrity) has been lowered from 30 seconds to 13 seconds.
Removing a sliver from the supermatter, the traitor objective, will further lower that down to 3 seconds.
Changes the wording on the mood effects from the supermatter delaminating slightly.
The crystal uses SPAN_COMMAND on its final countdown, which means it talk bigger.

## Why It's Good For The Game

Currently I feel that the supermatter delamination countdown overstays its welcome. Ideally it provides tension to get the hell out, or perhaps to make a risky last second play to try and save the supermatter. However right now its at 30 seconds, which gives no danger of staying in engineering right up to integrity 0, and keeps the tension at a high note for too long, almost to the point of awkwardness. 13 seconds is a good balance between get-the-fuck-out while still giving some leeway for engineers to escape and crewmembers to jump in lockers, and feels quick enough to give that danger that the supermatter should provide.
Additionally, removing a sliver from the supermatter lowers the cooldown to 3 seconds. Right now this is one of the harder tasks a traitor can be tasked with, while giving relatively little payoff sabatoge-wise. To the point where I have seen engineers just let the traitor do it, as the debuff it gives to the supermatter is minor. Now it makes the supermatter delaminate almost immediately after hitting 0 integrity, which means it will likely catch some engineers in the blast if a traitor did it stealthy. This also makes it more risky to try and fix a delamination if the engineering/security team did not stop the sliver from being removed. All meaning succeeding at this task should be more rewarding and damaging.
Finally the mood descriptions for the mood effects you get when a supermatter delaminates have been changed. Currently they are pretty gamey, and represent what the player might be thinking more than their character. Additionally they were not very descriptive of where they came from, which could be confusing.

## Changelog

:cl: Seven
balance: The supermatter delamination countdown has been lowered from 30 to 13 seconds
balance: Removing a sliver from the supermatter further lowers that down to 3 seconds
balance: The supermatter crystal uses bigger text on its final countdown
spellcheck: Some supermatter delamination related mood descriptions have been edited to explain the mood effect better
/:cl:

### CODE REVIEW

**Comment by iwishforducks** [comment]
the biggest issue with this, imo, is that the 30 countdown gave you ample time to see the SM whining in common and for you to finish what you’re doing and locker up. well, most of the time anyways. i still get caught out by the sm constantly.

getting permanent hallucinations because you didn’t get to a locker within 13 seconds as soon as the SM drops would feel pretty shitty.

**Comment by Lufferly** [comment]
> the biggest issue with this, imo, is that the 30 countdown gave you ample time to see the SM whining in common and for you to finish what you’re doing and locker up. well, most of the time anyways. i still get caught out by the sm constantly.
> 
> getting permanent hallucinations because you didn’t get to a locker within 13 seconds as soon as the SM drops would feel pretty shitty.

I like the idea that crewmembers need to rush to lockers, and may not even get one, but right now 30 seconds is a bit too long to make it feel urgent. 13 seconds makes it so you need to be quick, but it still should be doable for most crewmembers, and (I hope) will lead to clumsiness around lockers with multiple people trying getting in. Plus as long as im reading the code right the punishment for not getting in a locker is really not that bad, 100-600 seconds of hallucination depending on distance from the SM if you dont get in on time, plus a -2 mood effect for 4 minutes, or -8 and 5 minutes if its a cascade.

**Comment by lessthnthree** [comment]
Yeah, the hallucinations are tame. 30 seconds is too long and this should create an actual sense of urgency.

**Comment by Bibarra** [comment]
> the biggest issue with this, imo, is that the 30 countdown gave you ample time to see the SM whining in common and for you to finish what you’re doing and locker up. well, most of the time anyways. i still get caught out by the sm constantly.
> 
> getting permanent hallucinations because you didn’t get to a locker within 13 seconds as soon as the SM drops would feel pretty shitty.

They're not permanent and you can cure them with Synaptizine or other chems that stop hallucinations.

**Comment by iwishforducks** [comment]
> They're not permanent and you can cure them with Synaptizine or other chems that stop hallucinations.

wtf i thought the hallucinations were permanent! was that changed at some point or am i just stupid?

whatever that doesnt matter at all then. carry on with this pr then my good sir/ma’am/they’m

**Comment by Likteer** [comment]
The old moodlet was funny.
It does not need an explanation.

**Comment by SimplySmiley123** [comment]
> > the biggest issue with this, imo, is that the 30 countdown gave you ample time to see the SM whining in common and for you to finish what you’re doing and locker up. well, most of the time anyways. i still get caught out by the sm constantly.
> > getting permanent hallucinations because you didn’t get to a locker within 13 seconds as soon as the SM drops would feel pretty shitty.
> 
> I like the idea that crewmembers need to rush to lockers, and may not even get one, but right now 30 seconds is a bit too long to make it feel urgent. 13 seconds makes it so you need to be quick, but it still should be doable for most crewmembers, and (I hope) will lead to clumsiness around lockers with multiple people trying getting in. Plus as long as im reading the code right the punishment for not getting in a locker is really not that bad, 100-600 seconds of hallucination depending on distance from the SM if you dont get in on time, plus a -2 mood effect for 4 minutes, or -8 and 5 minutes if its a cascade.

You would be surprised about 30 seconds still somehow being too little for a lot of people.
13, or in worst case scenario just 10, means anyone who isn't already next to a locker is fucked. "Oh just listen to common for when the % drops low" Yeah and listen to engineering also yelling on common that they can fix it and all the other clutter.
And then comes people next to you who also may want to not get hallucinations that might as well be permament because nobody knows about synaptizine, will fuck over both you and themselves and often out of stupidity you had no control over.

**Comment by nukularpower** [comment]
Imo the entire delam process takes way too long, but this is a good start.  Would love to see the requirements for "special" delams made a little stricter

**Comment by nukularpower** [comment]
I've never even noticed that there were hallucinations from the SM delam and would argue that delam feels entirely too ignorable these days, with radiation axed.   If anything the hallucinations should be worse, though TG hallucinations in general are quite tame compared to most other servers

**Comment by distributivgesetz** [comment]
I wanna chime in, me being an engineering main, and give my thoughts as to why I feel like dropping the countdown this much would actually lessen the tension from supermatter delaminations.

The SM can recover from a countdown if you manage to heal the engine from 100% damage. It's pretty rare that this happens, you _really_ have to try hard to make sure the engine at least starts healing during the countdown, and that makes it all the more thrilling when you manage to get it out of the countdown. I and a lot of people have made it happen before. Staying behind in the engine room while the countdown is ongoing and wrenching shit around from a sabotage with those 30 seconds, with the hope that you can be the hero and save the engine is the experience I and a lot of other mains sign up for.

With this change I feel like it's gonna be way less likely, if not almost impossible to recover the engine from the countdown state, and that's really unfortunate. Everyone will have to evacuate from the engine room as soon as the countdown hits, because you won't have enough time to get the distance from the blast to do anything meaningful anymore. The countdown is an instant game over now.

This means that we will not hear `[Common] The supermatter crystal beeps, "Crystalline hyperstructure returning to safe operating parameters. Harmonic frequency restored within emergency bounds. Anti-resonance filter initiated."` likely at all anymore, and I think that that's unfortunate.

That's just my perspective and thoughts about this. I feel like it's valid sharing this though, since there hasn't been much talk about this part of the argument.

**Comment by newfren** [comment]
also sad about this making it significantly harder to save a delaminating supermatter

**Comment by Deadgebert** [comment]
I think a 13 second timer is too short. The issue of people getting to safety before it blows up is mostly irrelevant, most engineers will stay and fight a delam to the bitter end. That 30 second window is a good safety net and there's some bragging rights and a sense of satisfaction that comes with saving a delam during that period. I think we'd lose a lot of those last second attempts at saving with this change.

I love the idea of making the crystal easier to delam with an antag objective, I believe it's way too hard to delam in the current state of the game. Unfortunately that's an issue of atmospherics moving too slow later in the round, I don't know if changing the countdown itself would help much with that problem. I'd like to suggest an alternative, rather than reducing countdown time when a piece of the supermatter is removed, it might be more engaging if the integrity loss was changed instead. The integrity loss could be increased and the loss caps could be pushed back significantly.

**Comment by nukularpower** [comment]
IMO once its reached the 30 second countdown point, its fucked anyways - I dont think I've ever seen a delam recovered at that point personally, though I'm sure it's happened.    Anything to speed up the process is good in my mind

**Comment by Lufferly** [comment]
I do think that the supermatter is a little too hard to delam right now, but a balance change like that is probably best left to someone who understands the numbers better than me. To be honest, I don't really expect this to kill a ton more engineers, its really just to increase the dramatic effect the supermatter holds, and making it so the choice between staying and trying to fix the SM at 10% integrity is a more difficult choice.  As for the sliver SM objective, it honestly is still pretty tame, but it adds the intrigue that the supermatter can just explode in your face pretty much the moment it hits 0, and (I hope) will add a sense of "Ha! I did that! That was me!" to a traitor if the SM explodes. Im cautious about making the sliver modify the SM's numbers to sabatoge because A: For most players its gonna be pretty hidden and B: if a traitor really wants to delaminate the SM, they will end up in its room by virtue of doing the objective.

**Comment by RikuTheKiller** [comment]
Somehow, 13 bothers me way more than if it was 10 or 15.

**Comment by lessthnthree** [comment]
What if it was 15 seconds for a nice round number?

**Comment by bruhlookatthisdood** [comment]
i feel like when it delams it should give more of a warning than text now, like a global alarm or something.

**Comment by CandleJaxx** [comment]
i do think this would incentivize engineers to run away instead of trying to decon the floor to minimize the explosion range / potential singulos (which is good, more damage more fun and evil)

**Comment by Lufferly** [comment]
> i feel like when it delams it should give more of a warning than text now, like a global alarm or something.

Not a bad idea, I made it use bigger text (SPAN_COMMAND) on its final countdown

**Comment by Lufferly** [comment]
what

**Comment by JohnFulpWillard** [comment]
The board has been alerted ‼️

**Comment by Lufferly** [comment]
technical difficulties, please stand by

File: code/__DEFINES/tgs.dm [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // tgstation-server DMAPI
 
-#define TGS_DMAPI_VERSION "6.5.2"
+#define TGS_DMAPI_VERSION "6.5.3"
```
Comment (Cyberboss) [inline code/__DEFINES/tgs.dm:3]: This was already merged, why is GitHub showing it?

**Comment by Cyberboss** [comment]
I'd suggest rebasing to master tbh

**Comment by ZephyrTFA** [comment]
You messed up a master merge and rebased instead by accident.
You'll need to redo this pr, apologies

**Comment by Lufferly** [comment]
I am the joker now

**Pull request closed without merging**